spec/models/ability_spec.rb

```diff
@@ -105,4 +105,82 @@
     it { is_expected.not_to be_able_to(:update, FactoryBot.create(:field_set, project: project)) }
     it { is_expected.not_to be_able_to(:update, FactoryBot.create(:publish_target, project: project)) }
   end
+
+  describe 'for digital object permissions' do
+    let(:authorized_object) { FactoryBot.build(:digital_object_test_subclass, :with_sample_data, :with_lincoln_project) }
+    let(:unauthorized_object) { FactoryBot.build(:digital_object_test_subclass, :with_sample_data, :with_minken_project) }
+    let(:mixed_object) do
+      obj = FactoryBot.build(:digital_object_test_subclass, :with_sample_data)
+      obj.projects.merge authorized_object.projects
+      obj.projects.merge unauthorized_object.projects
+      obj
+    end
+    let(:authorized_project) { authorized_object.projects.to_a[0] }
+    let(:unauthorized_project) { unauthorized_object.projects.to_a[0] }
+
+    subject { described_class.new(user) }
+    context 'when a user has the ability to update_objects for a project' do
+      let(:user) do
+        FactoryBot.create(
+          :user, permissions: [
+            Permission.create(action: :update_objects, subject: Project.to_s, subject_id: authorized_project.id)
+          ]
+        )
+      end
+
+      it { is_expected.to be_able_to(:update, authorized_object) }
+      it { is_expected.to be_able_to(:update_objects, authorized_project) }
+      it { is_expected.not_to be_able_to(:update_objects, unauthorized_project) }
+      it { is_expected.to be_able_to(:update, mixed_object) }
+      it { is_expected.not_to be_able_to(:update, unauthorized_object) }
+      it { is_expected.not_to be_able_to(:destroy, authorized_object) }
+      it { is_expected.not_to be_able_to(:publish, authorized_object) }
+    end
+    context 'when a user has the ability to delete_objects for a project' do
+      let(:user) do
+        FactoryBot.create(
+          :user, permissions: [
+            Permission.create(action: :delete_objects, subject: Project.to_s, subject_id: authorized_project.id)
+          ]
+        )
+      end
+
+      it { is_expected.to be_able_to(:destroy, authorized_object) }
+      it { is_expected.to be_able_to(:delete_objects, authorized_project) }
+      it { is_expected.not_to be_able_to(:delete_objects, unauthorized_project) }
+      it { is_expected.to be_able_to(:destroy, mixed_object) }
```

Comment:
The way we would use this permissions is that they would only be deleted from within one of the projects that a digital object is in?

Reply:
The spec is deceptive because it's only in one project - my intention here was about object deletion and not relationship breaking. But should that be checked when an update removes a project? Tricky. Here's another question: Should this be more like create - should you have to have permission to delete in all the projects?

Reply:
If an object is in two projects, I don't think we should allow a user with only permission to delete from one project to delete it. Maybe we should treat a delete for one project as the relationship breaking and then actually delete the object if they aren't pointing to any more projects. In a way, the manage_all_digital_objects permissions encompasses the delete_all permission. Should only people with those permissions be able to delete objects?

Reply:
I agree, that requiring users to have permission to delete in all the projects an object is linked to should be sufficient. 

Yea, the relationship breaking could just be an update. Maybe just having a "Remove" button that sends off an update request. 

I still think we would want to delete an object if it's "Removed" from all the projects its a part of. Or say that a object can't be remove from the last project its a part of.

Reply:
The last-project question is a sticky one - part of me likes the idea of not letting the last one be removed, but... it also seems like a not-very-user-managed data at that point. I'm reluctant to delete when it isn't explicitly requested, but then should there be an admin-only orphans project?

Reply:
I don't like the idea of having objects that arent managed by a project, it could be hard to find out where an orphaned object came from and what should be done with it.

I think we should not allow the remove from the last project available, because at that point it should be a deletion and only the users with the correct permissions would have that option.